caasp-grafana-image/config.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+################################################################################
+# config.sh runs at the end of the prepare step, after users have been set,
+# packages installed, and the overlay filesystem created.
+################################################################################
+# http://osinside.github.io/kiwi/working_with_kiwi/shell_scripts.html
+# include Kiwi functions and vars
+test -f /.kconfig && . /.kconfig
+test -f /.profile && . /.profile
+
+OLDIFS="$IFS"  # to restore normal behavior
+DIRIFS="/$IFS" # to split directory components
+
+entrypoint="./usr/local/bin/entrypoint.sh"
+if [ -f "$entrypoint" ]
+then
+    Debug "Setting recursive permissions on entrypoint '$entrypoint'"
+    p=""
+    IFS="$DIRIFS"
+    for component in $entrypoint
+    do
+        IFS="$OLDIFS"
+        p="${p:-}${p:+/}$component"
+        Debug $(chmod -v 0755 "$p")
+        IFS="$DIRIFS"
+    done
+    IFS="$OLDIFS"
+else
+    Echo "Failed to find entrypoint '$entrypoint'; things will break"
```

Comment:
I think we should exit with an error, that should cause the KIWI build process to fail.

cc @davidcassany

Reply:
Yes I agree that better a failure than just some error messages. In any case I am concerned to take this config.sh approach, as it is already mentioned in SUSE/avant-garde#1770, this might be fixing a git to OBS workflow issue but no anything on the image itself. If this is the case this logic should not be part of the image and instead ensure that the tarballs created for images preserve the file attributes as committed in the git repository (related to SUSE/avant-garde#1801).

From a KIWI PoV modifying the file permissions of a file that is included as part of the overlay tree in config.sh is perfectly possible, but this is likely to be unnecessary as KIWI keeps the file permissions as they are already packed in the root.tar.gz tarball. It feels odd to me having to add a script to modify the permissions of our own custom files that are just being appended into the image by us.

My take is that if the procedure from git sources to the build inside OBS looses some file permissions control we should identify it rather than work around it.

Reply:
This logic needs to be in the process somewhere as part of listing things which need to be true in order for the image to function, like running a salt state or cloud init after provisioning a system. Anywhere earlier in the process is an opportunity for something to go wrong later along the line where we don't necessarily have control.

If OBS were capable of noting differences in permissions or ownership inside the tar file, we'd have a way to notice when this failed. But there is no such check yet. So kiwi is the only reasonable place I see to do that check before the image is built.

Reply:
Can we do both things:

  * Have the explicit failure introduced by @dannysauer to be sure our container image is going to work as expected
  * At the same time investigate the issue inside of OBS
  * Eventually, once the issue is fixed inside of OBS, drop the explicit failure

Reply:
Well the issue I'd say is not inside OBS, this is mostly part of the manual process involved in committing the new sources in OBS. I'd say the issue happens inside the committer host. Anyway, I am fine following this dual approach. I did not continue with the conversation on this topic as I expected during this sprint to start working on SUSE/avant-garde#1800, however due to v5 having higher priority it is still on the backlog and I am about to jump on vacation next week, thus it might remain in the backlog for a while. 

The mentioned card tries to provide PR tests in this repository by actually running the build inside OBS on a PR basis in a branched project, which also opens the door to include smoke tests per image basis against the build itself before actually merging merging the code. In that context I'd like to see the test that is suggesting @dannysauer. Since this might take a while I believe we can proceed with @flavio's suggestion as it might take a while before I manage to find the cycles to implement SUSE/avant-garde#1800.